contrib/utilities/run_clang_tidy.sh

```diff
@@ -41,9 +41,13 @@ if test ! -d "$SRC/source" -o ! -d "$SRC/include" -o ! -d "$SRC/examples" -o ! -
 fi
 echo "SRC-DIR=$SRC"
 
+# unset CC and CXX
+unset CC
+unset CXX
```

Comment:
@tjhei Completely unsetting CC and CXX fixes the configure issue. clang-tidy afterwards seems to be smart enough to accept the fact that the project had been configured with gcc.

Reply:
Alternatively, if you prefer, you can also add back the compiler wrappers (i.e. remove the two `unset`) and specify `-DMPI_INCLUDE_DIRS=/usr/include/x86_64-linux-gnu/mpich` as an option. This seems to work as well.

Reply:
@tjhei The current variant now uses the compiler wrappers + sets `MPI_INCLUDE_DIRS` manually to make clang-tidy happy.